contracts/contracts/lib/LibSubnetActor.sol

```diff
@@ -90,7 +92,7 @@ library LibSubnetActor {
                 emit SubnetBootstrapped(s.genesisValidators);
 
                 // register adding the genesis circulating supply (if it exists)
-                IGateway(s.ipcGatewayAddr).register{value: totalCollateral + s.genesisCircSupply}(s.genesisCircSupply);
```

Comment:
Looks like we had a bug here, where we sent native coins representing the genesis circulating supply, even if the supply source was ERC20. Nice catch!